src/modules/flow/wallclock/wallclock.c

```diff
@@ -573,9 +676,58 @@ static void
 wallclock_timeblock_close(struct sol_flow_node *node, void *data)
 {
     struct wallclock_timeblock_data *mdata = data;
+    int r;
 
     if (mdata->timer)
         sol_timeout_del(mdata->timer);
+
+    r = sol_platform_del_system_clock_monitor(system_clock_changed_timeblock,
+        mdata);
+    SOL_INT_CHECK(r, < 0, r);
+}
+
+static int
+wallclock_enabled_process(struct sol_flow_node *node, void *data, uint16_t port,
+    uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    bool enabled;
+    struct wallclock_data *mdata = data;
+
+    r = sol_flow_packet_get_boolean(packet, &enabled);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (enabled && !mdata->registered) {
+        struct wallclock_timer *timer = &timers[mdata->type];
+
+        r = wallclock_add_client(node, mdata);
+        SOL_INT_CHECK(r, < 0, r);
+        return sol_flow_send_irange_packet(node, 0, &timer->val);
+    }
```

Comment:
check if it was registered before we try to remove the client? Similar logic you did for timeblock.